src/UI/Components/Diagram/shapes.ts

```diff
@@ -244,24 +244,25 @@ export class ServiceEntityBlock extends shapes.standard.HeaderedRecord {
     return relations ? relations : null;
   }
 
-  addRelation(id: string, instanceName: string): void {
+  addRelation(id: string, relationName: string): void {
     const currentRelation = this.getRelations();
 
     if (currentRelation) {
-      this.set("relatedTo", currentRelation.set(id, instanceName));
+      this.set("relatedTo", currentRelation.set(id, relationName));
     } else {
       const relationMap = new Map();
-      relationMap.set(id, instanceName);
-      this.set("relatedTo", relationMap.set(id, instanceName));
+      this.set("relatedTo", relationMap.set(id, relationName));
     }
   }
 
-  removeRelation(id: string): void {
+  removeRelation(id: string): boolean {
```

Comment:
The name of the method is misleading, you want to know if it was sucessful or not, I get it, but the way you're using the method in the halo file, in an ifcase, is confusing.

Reply:
Hmm, what's your proposition? I based myself on delete() that is in the Map, but added this whole if()as some explanation, because delete returns the same information, but it could be more confusing without these extra few lines

Reply:
Why not work with a promises ? It would make more sense to me when using the removeRelation method. 

``` javascript
      if (relations) {
        didElementChange =
          didElementChange === true ||
          elementAsService.removeRelation(cellView.model.id as string).then(handle response)"
      }
```

Reply:
But why make it async?

Reply:
The way you wrote the method is similar to any sort of promise. Execute a task and then return if it was successful or not. That's a promise to me.

Reply:
I kind of agree, but in my opinion emphasis for promises should be on the fact of awaiting for that feedback. I wouldn't use actual Promise for that as that action in our case is completely synchronous.